### PR TITLE
Merge comments/edit & comments/form Partials

### PR DESCRIPTION
**Pull request by noi5e**
These two partials are basically very close copies of each other... Even more so with all the recent refactoring of the Comment Editor.

- [comments/_edit](https://github.com/publiclab/plots2/blob/main/app/views/comments/_edit.html.erb)
- [comments/_form](https://github.com/publiclab/plots2/blob/main/app/views/comments/_form.html.erb)

In the past, if I made a change in one partial, I had to do it in the other to match.

Merging these two partials into one, to increase maintainability and DRYness!

---
(This issue is part of the larger Comment Editor Overhaul Project with Outreachy. Refer to Planning Issue #9069 for more context)

### CODE REVIEW

File: app/views/comments/_form.html.erb [hunk 1]
```diff
@@ -1,104 +1,156 @@
+<!-- comment is false in reply & main comment forms; defined for edit forms -->
+<% comment_id_string = comment ? comment.id.to_s : nil %>
 <!-- get_comment_form_id defined in app/helpers/application_helper.rb: -->
-<% comment_form_id = get_comment_form_id(location, local_assigns[:reply_to], nil) %>
-<div class="comment-form-wrapper card-body" style="background-color:#f8f8f8; border: 1px solid #e7e7e7;padding: 18px;">
-  <form id="comment-form-<%= comment_form_id %>" class="comment-form" data-remote="true" action="/comment/create/<%= @node.nid %><%= "?type=question" if local_assigns[:type]=="question" %>" <% if local_assigns[:aid].blank? %>method="post"<% end %>>
-    <% if current_user && current_user.id == @node.uid %>
-      <span class="pull-right" style="color:#ccc;">
-        <a target="_blank" href="https://publiclab.org/comment-templates">Add a template</a>
-      </span>
```
Comment (noi5e) [inline app/views/comments/_form.html.erb:8]: Unrelated change: I'm deleting the template functionality in this PR.

File: app/views/comments/_form.html.erb [hunk 2]
```diff
@@ -1,104 +1,156 @@
...
+    padding: 18px;
+    <%= "display: none" if location == :edit  %>
+  "
+>
+  <form 
+    id="comment-form-<%= comment_form_id %>" 
+    class="comment-form" 
+    <% if location != :edit %>
+      action="/comment/create/<%= @node.nid %>"
+      data-remote="true"
```
Comment (noi5e) [inline app/views/comments/_form.html.erb:20]: This `data-remote="true"` (along with `comment.js`) seems to play a crucial role in form submission in edit forms vs. the others.

Edit comment submissions refresh the page, while reply & main comment form submissions don't!

Funnily enough, this was something that was stumping me in [this comment here](https://github.com/publiclab/plots2/issues/8810#issuecomment-747867020) way back a the beginning of the internship.

Comment (noi5e) [reply app/views/comments/_form.html.erb:20]: Like, if you're just searching in VSCode for places where [submitFormAjax](https://github.com/publiclab/plots2/blob/main/app/assets/javascripts/submitFormAjax.js) and [addComment](https://github.com/publiclab/plots2/blob/main/app/assets/javascripts/submitFormAjax.js) are called, you don't find anything. For the longest time, I thought that these functions weren't actually being called.

But now I think they are! They work based on attaching extra classes to particular comment forms. I think. This intrigues me because a similar thing could be done to edit comment forms so they are submitted with AJAX and don't refresh the page.

Comment (jywarren) [reply app/views/comments/_form.html.erb:20]: OK, i think i can help here -- the `data-remote` attribute is used by the obscure `rails-ujs` library -- https://guides.rubyonrails.org/working_with_javascript_in_rails.html#remote-elements

However this is so obscure that I think it's just not worth using this technique; we don't have to remove it but let's add an explanatory comment to help future coders.

Comment (noi5e) [reply app/views/comments/_form.html.erb:20]: Ahh, very good to know!

File: app/views/comments/_form.html.erb [hunk 3]
```diff
@@ -1,132 +1,217 @@
+<!-- comment is false in reply & main comment forms; defined for edit forms -->
+<% comment_id_string = comment ? comment.id.to_s : nil %>
 <!-- get_comment_form_id defined in app/helpers/application_helper.rb: -->
-<% comment_form_id = get_comment_form_id(location, local_assigns[:reply_to], nil) %>
-<div class="comment-form-wrapper card-body" style="background-color:#f8f8f8; border: 1px solid #e7e7e7;padding: 18px;">
-  <form id="comment-form-<%= comment_form_id %>" class="comment-form" data-remote="true" action="/comment/create/<%= @node.nid %><%= "?type=question" if local_assigns[:type]=="question" %>" <% if local_assigns[:aid].blank? %>method="post"<% end %>>
```
Comment (noi5e) [inline app/views/comments/_form.html.erb:4]: Deleting this `?type=question` part here.

**Comment by noi5e** [comment]
Just did a lot of double-checking of the now-deleted edit partial. This one is ready to merge!

**Review by cesswairimu — APPROVED** [review]
This is looking great :rocket:  thanks for working on this

**Review by jywarren — APPROVED** [review]

**Comment by jywarren** [comment]
This looks great. Thank you! This will also be easy to track if there are any errors, as the PR touches both files. So even if we catch something later this will be traceable. Thanks a lot!!!
